lib/Twig/BaseNodeVisitor.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Twig_BaseNodeVisitor can be used to make node visitors compatible with Twig 1.x and 2.x.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+abstract class BaseNodeVisitor implements Twig_NodeVisitorInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public function enterNode(Twig_NodeInterface $node, Twig_Environment $env)
+    {
+        return $this->enterNode($node, $env);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    final public function leaveNode(Twig_NodeInterface $node, Twig_Environment $env)
+    {
+        return $this->leaveNode($node, $env);
+    }
+
+    abstract public function enterNode(Twig_Node $node, Twig_Environment $env);
```

Comment:
this is broken. It must use a different name than the public method.

and it should be a protected method